roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/RoundaboutCar.java

```diff
@@ -65,22 +75,27 @@ public double getTimeToTraverseCurrentSection() {
     }
 
     @Override
-    public double getTimeToTraverseSection(IStreetSection section) {
+    public double getTimeToTraverseSection(Street section) {
 
-        double carPosition = 0;
+        if (section instanceof StreetSection) {
+            double carPosition = 0;
 
-        if (section.getCarPositions().containsKey(this)) {
-            carPosition = section.getCarPositions().get(this);
-        }
+            if (section.getCarPositions().containsKey(this)) {
+                carPosition = section.getCarPositions().get(this);
+            }
 
-        double remainingLength = section.getLength() - carPosition;
-        return remainingLength / this.getDriverBehaviour().getSpeed();
+            double remainingLength = section.getLength() - carPosition;
+            return remainingLength / this.getDriverBehaviour().getSpeed();
+        } else if (section instanceof OneWayStreetAdapter) {
+            return 0;
+        }
+        return -1;
```

Comment:
throw new IllegalStateException("street needs to instance of streetsection or onewaystreetadapter")  ?